### PR TITLE
Add libsynctex

### PR DESCRIPTION
**Pull request by mvwieringen**
Needed for building Mate 1.24 atril version

### CODE REVIEW

File: components/library/libsynctex/Makefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#
+
+BUILD_BITS=             64
+BUILD_STYLE=		justmake
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		libsynctex
+COMPONENT_VERSION=      2.0.0
+COMPONENT_REVISION=	0
```
Comment (AndWac) [inline components/library/libsynctex/Makefile]: Remove COMPONENT_REVISION. We set it only if needed (starting with 1).

File: components/library/libsynctex/Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+COMPONENT_PROJECT_URL=  https://github.com/jlaurens/synctex
+COMPONENT_LICENSE=	MIT
+COMPONENT_LICENSE_FILE=	LICENSE
+
+TEST_TARGET=            $(NO_TESTS)
+
+include $(WS_MAKE_RULES)/common.mk
+
+PATH=/usr/gnu/bin:/usr/bin
+
```
Comment (AndWac) [inline components/library/libsynctex/Makefile:42]: PATH=$(PATH.gnu)

Comment (AndWac) [inline components/library/libsynctex/Makefile:43]: PATH=$(PATH.gnu)

**Comment by pyhalov** [comment]
I have 2 questions:
1) Should we provide complete texlive instead?
2) In case we build atril without this library, what is missing?

**Comment by mvwieringen** [comment]
> I have 2 questions:
> 
> 1. Should we provide complete texlive instead?
Honestly I think providing texlive completely is mostly overkill as atril only needs the parser I just packaged just enough so
not needing to have a big dependency.

> 2. In case we build atril without this library, what is missing?
As far as I remember you cannot build without it but I could be wrong. The actual thing is that the parser I now
packaged used to be part of atril as bundled source but they decided as most Linux distributions deliver it as
part of texlive to make it an external dependency. I now also added the latest version from the original author of
the parser which is in a different repo then texlive. (The only disadvantage is that in that repo there is no build
system so I had to hack a rudimentary Makefile but given that the whole parser is 2 source files and some headers
the current solution is the most lightweight and simplest solution for providing atril with its needed now external source
for the parser and not needing to package the full texlive stuff.)

**Comment by AndWac** [comment]
I have been thinking about providing texlive like OmniOSce is doing. It is on my wishlist for some time but I haven't found enough time for it.

**Comment by mvwieringen** [comment]
@AndWac this one also has to merge before mate 1.24 can merge. But you can always drop this stuff and replace it with the full texlive in the future.

**Pull request merged**